executor/internal/mpp/local_mpp_coordinator.go

```diff
@@ -275,6 +313,22 @@ func (c *localMppCoordinator) fixTaskForCTEStorageAndReader(exec *tipb.Executor,
 	return nil
 }
 
+// DFS to check if plan needs report execution summary through ReportMPPTaskStatus mpp service
+// Currently, return true if plan contains limit operator
+func needReportExecutionSummary(plan plannercore.PhysicalPlan) bool {
+	switch x := plan.(type) {
+	case *plannercore.PhysicalLimit:
```

Comment:
Also for TopN

Reply:
Does our TopN operator implementation terminate mpp task before receiving all data now?

Reply:
yes, topn pushed down to tiflash is partial topn, as follows
```
tidb         tiflash
topX <--  topX
                topX
                topX
```
The total number of rows passed from tiflash to tidb will be tiflash_node * X, but tidb closes the executor after receiving X rows.

Reply:
My fault, topn won't

Reply:
all right~